library/src/main/java/com/nextcloud/common/NextcloudClient.kt

```diff
@@ -118,6 +119,11 @@ class NextcloudClient private constructor(
         return method.execute(this)
     }
 
+    @Throws(Exception::class)
+    fun execute(method: OkHttpDavMethodBase) {
```

Comment:
Ideally we'd have the dav method extend from `OkHttpMethodBase` as well, and `NextcloudClient` would not care about the kind of method it's implementing.

I guess this is due to the different execution model of `dav4jvm` (which doesn't let you access the methods it creates, and instead executes them directly).

If this is not possible, then we should have `OkHttpMethodBase` and `OkHttpDavMethodBase` at least implement a common interface (like `NextcloudClientMethod` or something) and have a single `execute` method that takes that interface.